wp-includes/class-wp-https-detection.php

```diff
@@ -107,14 +133,93 @@ public function check_https_support() {
 			return $response;
 		}
 
-		if ( false === strpos( wp_remote_retrieve_body( $response ), wp_hash( $this->query_var_number, 'nonce' ) ) ) {
+		$body = wp_remote_retrieve_body( $response );
+
+		if ( ! $this->has_proper_manifest( $body ) ) {
 			return new WP_Error(
 				'invalid_https_validation_source',
-				__( 'There was an issue in the request for HTTPS verification. It might not have been from the same origin.', 'default' )
+				__( 'There was an issue in the request for HTTPS verification.', 'pwa' )
 			);
 		}
 
-		return 200 === wp_remote_retrieve_response_code( $response );
+		return $response;
+	}
+
+	/**
+	 * Gets whether the body has a manifest <link>, with the proper href value.
+	 *
+	 * @param string $body The body of the response.
+	 * @return boolean Whether the body has a proper manifest.
+	 */
+	public function has_proper_manifest( $body ) {
+		$libxml_previous_state = libxml_use_internal_errors( true );
```

Comment:
This prevents errors, like in [parsing HTML5 elements](http://php.net/manual/es/domdocument.loadhtml.php#118107).